apis/v1alpha1/destination.go

```diff
@@ -143,19 +138,51 @@ func (dest *Destination) GetRef() *corev1.ObjectReference {
 	return nil
 }
 
-func validateDestinationRef(ref corev1.ObjectReference) *apis.FieldError {
-	// Check the object.
+func IsValidObjectReference(f corev1.ObjectReference) *apis.FieldError {
+	return checkRequiredObjectReferenceFields(f).
+		Also(checkDisallowedObjectReferenceFields(f))
+}
+
+// Check the corev1.ObjectReference to make sure it has the required fields. They
+// are not checked for anything more except that they are set.
+func checkRequiredObjectReferenceFields(f corev1.ObjectReference) *apis.FieldError {
 	var errs *apis.FieldError
-	// Required Fields
-	if ref.Name == "" {
+	if f.Name == "" {
 		errs = errs.Also(apis.ErrMissingField("name"))
 	}
-	if ref.APIVersion == "" {
+	if f.APIVersion == "" {
 		errs = errs.Also(apis.ErrMissingField("apiVersion"))
 	}
-	if ref.Kind == "" {
+	if f.Kind == "" {
 		errs = errs.Also(apis.ErrMissingField("kind"))
 	}
-
 	return errs
 }
+
+// Check the corev1.ObjectReference to make sure it only has the following fields set:
+// Name, Kind, APIVersion
+// If any other fields are set and is not the Zero value, returns an apis.FieldError
+// with the fieldpaths for all those fields.
+func checkDisallowedObjectReferenceFields(f corev1.ObjectReference) *apis.FieldError {
+	disallowedFields := []string{}
+	// See if there are any fields that have been set that should not be.
+	s := reflect.ValueOf(f)
+	typeOf := s.Type()
+	for i := 0; i < s.NumField(); i++ {
+		field := s.Field(i)
+		fieldName := typeOf.Field(i).Name
+		if fieldName == "Name" || fieldName == "Kind" || fieldName == "APIVersion" {
```

Comment:
(Optional) Switch?

Reply:
yeah, copy/paste from eventing.. can change it, but I think we won't end up merging this